src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react';
+
+export const useLocalStorage = <T>(key: string, initialValue: T): [T, (value: T) => void] => {
```

Comment:
I'm not super keen on this hook.

- This hook will always set a default value. Even if we change the default value at some point, the user will still be stuck on the old default (without making an active choice).

- The hook has no context/listeners to sync value change between components.
![Recording 2024-11-05 102209](https://github.com/user-attachments/assets/bdcb1d1d-acce-4077-8b46-e22c9bf7e584)


I'd rather see us using `usehooks-ts`.
Note: even if we'd use `usehooks-ts`, I'd suggest to use the option `initializeWithValue: false`

I could live with this hook existing if:
- keys were namespace (eg `grafana.syntheticMonitoring[<instanceId>].<key>`)
- There were comments (pref JSDoc) clearly stating that state is not synced
- `initialValue` changed name to `defaultValue`, with an option `{setDefaultValue: Boolean = false}` (which would return the default value without writing it to localStorage).
- (feature request 😅) if the hook would would allow the developer to select `local|session` storage with an option/param eg. `useClientStorage<T>(key: string, defaultValue?: T, options?: ClientStorageOptions)`

Reply:
Added `usehooks-ts` instead, love the set of hooks it provides!